src/nanoarrow/meson.build

```diff
@@ -80,6 +80,34 @@ if get_option('ipc')
                                            dependencies: [nanoarrow_dep, flatcc_dep])
 endif
 
+needs_device = get_option('device') or get_option('metal') or get_option('cuda')
+if needs_device
+  device_deps = [nanoarrow_dep]
+  device_srcs = ['nanoarrow_device.c']
+
+  if get_option('metal')
+    metal_dep = dependency('appleframeworks', modules: ['Foundation', 'Metal'])
+    metal_cpp_dep = dependency('metal-cpp')
+    device_deps += metal_dep
+    device_deps += metal_cpp_dep
+    device_srcs += 'nanoarrow_device_metal.cc'
+  endif
+
+  if get_option('cuda')
+    cuda_dep = dependency('cuda', modules: ['cuda_driver'])
```

Comment:
Just guessing on the module name. Hoping it matches what CMake has (and what exists in our current CMake config)

https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html